### ISSUE TITLE
[BUG] checkIfDefined is doing nothing but wasting CPU

### ISSUE THREAD

**Issue by Gusted** [darkreader/darkreader#3905]
### **Describe the bug**

The current implementation can and is endlessly looping so long the elements are not defined which in some cases won't be and due to the logic used in this function it will somehow use noticeable (on low-medium spec CPU) CPU Utilization.

e.g. https://mail.google.com

### **Steps to reproduce**

- Go to mail.google.com
- Open performance tab in Dev tools
- Get a fresh performance `ctrl +e` after the page has been fully loaded.
- See that even that checkIfDefined is still being called in a animationFrame alongside checkForUpdate(expected)

### **Expected behavior**

Not using CPU or some other implementation to prevent this, not even sure what `:defined` has to do with Shadow DOM's.

### **Actual behavior**

It's using CPU utilization even in `idle` when not needed

### **System Information:**
 - OS: Windows 10
 - Browser: Vivaldi(3.4.2066.76)
 - Dark Reader version: 4.9.21

### **Additional context**

More info about this https://github.com/darkreader/darkreader/issues/3811#issuecomment-712695274

**Comment by Gusted** [comment]
@alexanderby What's the exact purpose of the defined Elements implementation?

**Comment by ddnexus** [comment]
@Gusted glad that you already pointed out what is happening! 👍

**Comment by Gusted** [comment]
It's a wild guess that this could be the problem, but this is a problem on its own that this is endlessly calling itself it should reduce your idle CPU usage, as for the checkForUpdate, maybe we can switch to some hackery in order to detect changes on another passive way that way the idle of Dark reader is 100% 0.0 if nothing happens from the web page of course

**Comment by ddnexus** [comment]
> on low-medium spec CPU

FYI, it happens with a 16 core Intel® Xeon(R) W-10885M CPU running at 3.5+GHz which is quite fast actually.
If it does not happen with other hardware, it could even be related to be too fast. 

I got that kind of problem a few times with fast executed javascripts on other platform. Sometimes the previous line may not get the change in the DOM fast enough before the second line kicks in not finding the change yet, causing weird errors. I remember that I had to use `setTimeout` to artificially delay the execution in some case.

Not sure if that could be the case... just a wild guess :)
